plugins/inputs/fritzbox/fritzbox_test.go

```diff
@@ -0,0 +1,178 @@
+package fritzbox
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/logger"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"github.com/tdrn-org/go-tr064/mock"
+)
+
+func TestInitDefaults(t *testing.T) {
+	plugin := defaultFritzbox()
+	err := plugin.Init()
+	require.NoError(t, err)
+	require.Len(t, plugin.URLs, 0)
+	require.Equal(t, []string{"device", "wan", "ppp", "dsl", "wlan"}, plugin.Collect)
+	require.Equal(t, defaultTimeout, plugin.Timeout)
+	require.Empty(t, plugin.TLSCA)
+	require.Empty(t, plugin.TLSCert)
+	require.Empty(t, plugin.TLSKey)
+	require.Empty(t, plugin.TLSKeyPwd)
+	require.False(t, plugin.InsecureSkipVerify)
+	require.NotNil(t, plugin.Log)
+}
+
+func TestConfig(t *testing.T) {
+	conf, err := os.ReadFile("testdata/conf/fritzbox.conf")
+	require.NoError(t, err)
+	var plugin = defaultFritzbox()
+	err = toml.Unmarshal(conf, plugin)
+	require.NoError(t, err)
```

Comment:
Probably better to use `config.LoadConfigData`.

Reply:
Adapted via 21f1990b168ddd6b189fe4f52cc932aa6cd31ceb